base/download.jl

```diff
@@ -26,15 +26,12 @@ if Sys.iswindows()
 else
     function download(url::AbstractString, filename::AbstractString)
         global downloadcmd
+        download_agents = ("curl", "wget", "fetch")
         if downloadcmd === nothing
-            for checkcmd in ("curl", "wget", "fetch")
-                try
-                    # Sys.which() will throw() if it can't find `checkcmd`
-                    Sys.which(checkcmd)
-                    downloadcmd = checkcmd
-                    break
-                catch
-                end
+            try
+                downloadcmd = basename(something(Sys.which.(download_agents)...))
+            catch
+                error("no download agent available; install curl, wget, or fetch")
             end
```

Comment:
Would be better (and not longer) to write this as a loop, rather than obscuring all errors with this message

Reply:
What else can throw here except `something(nothing)`?

Reply:
I agree it would be nicer to do without try/catch entirely, since that's now possible here.